packages/transition-frontend/src/components/forms/comparison/ScenarioComparisonPanel.tsx

```diff
@@ -295,7 +295,11 @@ class ScenarioComparisonPanel extends ChangeEventsForm<ComparisonPanelProps, Com
     private getScenarioNameById(id: string): string {
         for (const scenario of this.state.scenarioCollection.features) {
             if (scenario.id === id) {
-                return scenario.toString(false);
+                let scenarioName = scenario.toString(false) as string;
+                if (scenarioName.length >= 30) {
```

Comment:
Should the 30 be configurable ? (You probably want to put it in a constant anyway to make sure we use the same one in the if and in the substr in the next line)